test/images/agnhost/Dockerfile

```diff
@@ -20,15 +20,28 @@ CROSS_BUILD_COPY qemu-QEMUARCH-static /usr/bin/
 # install necessary packages:
 # - curl, nc: used by a lot of e2e tests
 # - iproute2: includes ss used in NodePort tests
-RUN apk --update add curl netcat-openbsd iproute2 && rm -rf /var/cache/apk/*
+# from iperf image
+# install necessary packages: iperf, bash
+RUN apk --update add curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/* \
+  && ln -s /usr/bin/iperf /usr/local/bin/iperf \
+  && ls -altrh /usr/local/bin/iperf
 
 # PORT 8080 needed by: netexec, nettest
 # PORT 8081 needed by: netexec
-EXPOSE 8080 8081
+# PORT 9376 needed by: serve-hostname
+EXPOSE 8080 8081 9376
 
 # from netexec
 RUN mkdir /uploads
 
+# from porter
+ADD porter/localhost.crt localhost.crt
+ADD porter/localhost.key localhost.key
+
 ADD agnhost agnhost
+
+# needed for the entrypoint-tester related tests.
+RUN ln -s agnhost agnhost-2
```

Comment:
this one's not super obvious to me atm, looking at the existing entrypoint-tester docker file

Reply:
This is probably fine.  It's subtly different from the old image in that this is a symlink, not an entirely separate binary.  This is why printing `os.Args` is important, so we can see how this was invoked to prove that a different command was used.

Reply:
That is correct. TBH, I was also confused initially confused about why the ``entrypoint-tester`` image had the binary added twice, until I looked at the tests. I added a small comment here stating that it's actually needed by those tests, but I should probably add more details about it.

And yeah, it's a symlink in order to save space. Adding the same binary again would almost double the image size, while the same effect can be achieved with a symlink.